src/main/java/dev/jbang/cli/Alias.java

```diff
@@ -79,13 +74,15 @@ public Integer doCall() {
 					"Invalid alias name, it should start with a letter followed by 0 or more letters, digits, underscores or hyphens");
 		}
 		try {
-			Script script = BaseScriptCommand.prepareScript(scriptOrFile);
+			ExtendedRunUnit script = RunUnit.forResource(scriptOrFile);
 			if (name == null) {
 				name = AppInstall.chooseCommandName(script);
 			}
 
-			if (description == null && script.getDescription() != null && script.getDescription().length() > 0) {
-				description = script.getDescription();
+			if (description == null && script.runUnit instanceof Script) {
```

Comment:
why need instanceof to do things right ? that feels wrong...

Reply:
Why? It's functionality that is only supported by Scripts. To me it seems more wrong to add a `getDescription()` to Jar, where it doesn't make any sense, just to avoid an `instanceof`?

Reply:
I've added the `getDescription()` to `RunUnit`, who knows JAR files might have a description at some point.